src/main/java/uk/gov/hmcts/divorce/payment/PaymentService.java

```diff
@@ -186,6 +189,10 @@ private PbaResponse getPbaErrorResponse(String pbaNumber, FeignException excepti
             return new PbaResponse(httpStatus, String.format(NOT_FOUND.value(), pbaNumber), null);
         }
 
+        if (isGenericErrorRequiredForHttpStatus(httpStatus)) {
+            return new PbaResponse(httpStatus, ERROR_GENERIC, pbaNumber);
+        }
+
```

Comment:
Will triggered a scenario where the internal feign client for fee-and-pay isn't reachable. If that scenario is possible as well (which is likely), we will need to bring the check for the httpStatus(500, 504) codes right after the FeignException is caught to return the generic error before other conditions are checked for other possible error conditions.